### DISCUSSION TITLE
Syncing a TTL with grating series using BonVision

### DISCUSSION THREAD

**Discussion by LordKelv** [bonsai-rx/bonsai#1710] [Q&A] [answered]
¡Hola Bonsai Rx gang!

I followed [these](https://bonvision.github.io/pages/08-GratingSeries/#) instructions to make a series of flashing gratings.  Here's the Bonsai Rx code for it: [gratings.zip](https://github.com/bonvision/BonVision/files/14561562/gratings.zip).  Here's a screenshot of the workflow as well.
![image](https://github.com/bonvision/BonVision/assets/6320889/65ad8a5a-bccf-426d-97ae-757160bdb955)
I would like for an Arduino to send a TTL pulse as the grating is showed, to sync it with an ephys rig.  

I cannot figure out where to insert the digital output module to have it flag every single time a grating is showed (once every second for 5 minutes).

I also [cross-posted](https://github.com/bonvision/BonVision/issues/30) this issue on the BonVision github. 

Many thanks,
Kelvin

**Comment by bruno-f-cruz** [comment, marked as answer]
Hi,
The toggle time of the digital output will not be aligned to the presentation of the visual stimuli as these run on two different systems (Arduino and a Display). If you want to precisely align the onset of the stimulation, you should use a photodiode that measures the actual physical change on your screen.
